iib/workers/tasks/opm_operations.py

```diff
@@ -26,29 +31,194 @@
 log = logging.getLogger(__name__)
 
 
-def _gen_port_for_grpc() -> Generator[int, None, None]:
+class PortFileLock:
+    """A class representing file-lock used during OPM operations."""
+
+    def __init__(self, purpose: str, port: int, lock_on_init: bool = False):
+        """Initialize the PortFileLock object."""
+        self.purpose = purpose
+        self.port = port
+        self.locked = False
+        self.filename = os.path.join(
+            tempfile.gettempdir(),
+            f'iib_{purpose}_{port}.lock',
+        )
+        if lock_on_init:
+            self.lock_acquire()
+
+    def lock_acquire(self):
+        """
+        Create a file representing port lock.
+
+        Before trying to create a a port-lock file, we try to check if the port is free.
+        """
+        log.debug(f"Attempt to lock port {self.port}.")
+
+        if self.locked:
+            log.debug(f"Error: Port {self.port} is already locked")
+            raise IIBError("Port lock attempt on already locked port")
+
+        # check if the port is free, opm service is doing the check too,
+        # however this way, we do not have to rely on their error message format
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        try:
+            s.bind(("localhost", self.port))
```

Comment:
You have to also close the port before calling opm.

Reply:
What about this one? Should we close the port?

Reply:
Fixed, please check